conjurapi/variable.go

```diff
@@ -14,25 +15,39 @@ import (
 //
 // The authenticated user must have execute privilege on all variables.
 func (c *Client) RetrieveBatchSecrets(variableIDs []string) (map[string][]byte, error) {
-	resp, err := c.retrieveBatchSecrets(variableIDs)
+	jsonResponse, err := c.retrieveBatchSecrets(variableIDs, false)
 	if err != nil {
 		return nil, err
 	}
 
-	data, err := response.DataResponse(resp)
-	if err != nil {
-		return nil, err
+	resolvedVariables := map[string][]byte{}
+	for id, value := range jsonResponse {
+		resolvedVariables[id] = []byte(value)
 	}
 
-	jsonResponse := map[string]string{}
-	err = json.Unmarshal(data, &jsonResponse)
+	return resolvedVariables, nil
+}
+
+// RetrieveBatchSecrets fetches values for all variables in a slice using a
+// single API call. This version of the method will automatically base64-encode
+// the secrets on the server side allowing the retrieval of binary values in
+// batch requests.
+//
+// The authenticated user must have execute privilege on all variables.
+func (c *Client) RetrieveBase64BatchSecrets(variableIDs []string) (map[string][]byte, error) {
```

Comment:
just to play devil's advocate, is it clear enough that this method doesn't return base64-encoded secrets?

I can imagine thinking `RetrieveBase64BatchSecrets` will return encoded secrets.

if you agree this might be a point of confusion, can we make the comment clearer on this point?

Reply:
I agree with you that the name of the method was confusing. I renamed it to `RetrieveBatchSecretsSafe` which I think reflects the usage better and added an extra comment in the doctring.